src/update/bin/steps.js

```diff
@@ -452,8 +452,74 @@ const askUnpaywall = async (jobConfig) => {
   return snapshotsInfo;
 };
 
+/**
+ * Start the download of the big file from unpaywall
+ * @param {String} stateName - state filename
+ * @param {String} info - information of the file to download
+ */
+const downloadBigSnapshot = async (stateName, info) => {
+  const { filename } = info;
+
+  const filepath = path.resolve(snapshotsDir, filename);
+
+  await addStepDownload(stateName);
+  const state = await getState(stateName);
+  const step = state.steps[state.steps.length - 1];
+  step.file = filename;
+  await updateStateInFile(state, stateName);
+
+  let res;
+  try {
+    res = await unpaywall({
+      method: 'get',
+      url: '/feed/snapshot/',
+      responseType: 'stream',
+      params: {
+        api_key: apiKey,
+      },
+    });
+  } catch (err) {
+    logger.error(`Cannot request ${unpaywall.baseURL}/feed/snapshot&api_key=${apiKey}`);
+    logger.error(err);
+    await fail(stateName, err);
+    return false;
+  }
+
+  info.size = res.headers['content-length'];
+
+  if (res?.data instanceof Readable) {
+    await new Promise((resolve, reject) => {
+      const writeStream = res.data.pipe(fs.createWriteStream(filepath));
+
+      const start = new Date();
+      updatePercentStepDownload(filepath, info, stateName, start);
+
+      writeStream.on('finish', async () => {
+        step.status = 'success';
+        step.took = (new Date() - start) / 1000;
+        step.percent = 100;
+        state.steps[state.steps.length - 1] = step;
+        await updateStateInFile(state, stateName);
+        logger.info('step - end download');
+        return resolve();
+      });
+
+      writeStream.on('error', async (err) => {
+        logger.error(err);
+        await fail(stateName, err);
+        return reject(err);
+      });
+    });
+  } else {
+    const writeStream = await fs.createWriteStream(filepath);
+    writeStream.write(res.data);
+    writeStream.end();
```

Comment:
```suggestion
   try {
     await fs.writeFile(filepath, res.data);
   } catch (e) {
     await fail(stateName, err);
     return reject(e);
   }
```

Looks like you're just writing content to a file, so `fs.writeFile()` may be simpler. It's also easier to handle the error that way.